### PR TITLE
Type checks before contract call in Ethereum

### PR DESCRIPTION
**Pull request by rranjan3**
Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

### CODE REVIEW

File: sdk/avalon_sdk/connector/blockchains/ethereum/ethereum_worker_registry.py [hunk 1]
```diff
@@ -217,18 +221,18 @@ def worker_register(self, worker_id, worker_type, organization_id,
         """
 
         if (self.__contract_instance is not None):
-            if not is_valid_hex_str(worker_id):
+            if not is_valid_hex_of_length(worker_id, 32):
                 logging.error("Invalid worker id {}".format(worker_id))
                 return None
```
Comment (Ram-srini) [inline sdk/avalon_sdk/connector/blockchains/ethereum/ethereum_worker_registry.py:226]: Return ContractResponse.ERROR instead of None for all invalid inputs if it is write transaction to chain.

File: sdk/avalon_sdk/connector/blockchains/ethereum/ethereum_work_order.py [hunk 2]
```diff
@@ -114,6 +114,15 @@ def work_order_submit(self, work_order_id, worker_id, requester_id,
         """
         if (self.__contract_instance is not None):
 
+            if not is_valid_hex_of_length(work_order_id, 32):
```
Comment (manojgop) [inline sdk/avalon_sdk/connector/blockchains/ethereum/ethereum_work_order.py:117]: I would have preferred to add all these constants lengths in a separate file

File: tests/test_utility/test_hex_utils.py [hunk 3]
```diff
@@ -75,6 +76,48 @@ def test_hex_to_byte_arrary(self):
...
+        self.assertFalse(is_valid_hex_of_length(str))
+
+        str = "0X12345ab"  # Positive case for length (<)
+        self.assertFalse(is_valid_hex_of_length(str, 16))
+
+        str = "0X12345abcde12345a"  # Positive case for length (==)
+        self.assertTrue(is_valid_hex_of_length(str, 16))
+
+        str = "0X12345abcde12345a"  # Negative case for length (>)
+        self.assertFalse(is_valid_hex_of_length(str, 8))
```
Comment (pankajgoyal2) [inline tests/test_utility/test_hex_utils.py:119]: you can add some tests to check for negative values of length

File: config/kme_config.toml [hunk 4]
```diff
@@ -81,8 +81,9 @@ ReceiptInvocationAddress = ""
 # Ethereum Address used by the worker to submit transactions
 FromAddress = ""
 # Replace ApplicationTypeId and OrganizationId with real hexadecimal values
-ApplicationTypeId = "11aa22bb33cc44dd"
-OrganizationId = "aabbcc1234ddeeff"
+ApplicationTypeId = "11aa22bb33cc44dd11aa22bb33cc44dd11aa22bb33cc44dd11aa22bb33cc44dd"
```
Comment (manju956) [inline config/kme_config.toml:84]: Is ApplicationTypeId need to be hex string of length 64 chars? If so, can you add a comment?
Same applies to OrganizationId as well.
